src/globus_sdk/local_endpoint/server/endpoint.py

```diff
@@ -0,0 +1,84 @@
+import json
+import pathlib
+import typing as t
+
+
+class LocalGlobusConnectServer:
+    r"""
+    A LocalGlobusConnectServer object represents the available SDK methods
+    for inspecting and controlling a running Globus Connect Server
+    installation.
+
+    These objects do *not* inherit from BaseClient and do not provide methods
+    for interacting with any Globus Service APIs.
+
+    :param info_path: The path to the info file used to inspect the local system
+    :type info_path: str, optional
+    """
+
+    def __init__(
+        self,
+        *,
+        info_path: t.Union[
+            str, pathlib.Path
+        ] = "/var/lib/globus-connect-server/info.json"
+    ) -> None:
+        self.info_path = pathlib.Path(info_path)
+        self._loaded_info: t.Optional[t.Dict[str, t.Any]] = None
+
+    @property
+    def info_dict(self) -> t.Optional[t.Dict[str, t.Any]]:
+        """
+        The info.json data for the local Globus Connect Server, as a dict.
+
+        If the info.json file is not present or cannot be parsed, the data is None.
+        This indicates that there is no local Globus Connect Server node, or if there is
+        one, it cannot be used or examined by the SDK. For example, containerized
+        applications using the SDK may not be able to interact with Globus Connect
+        Server on the container host.
+
+        :type: dict
```

Comment:
Should the annotations throughout (like this one) reflect that None is a failure-case possibility?

Reply:
I was on the fence about that, so I'm glad you brought it up. If the type is documented as `X or None`, is that sufficient? Or should each one also have a sentence in the docstring about "if the value cannot be read, this is `None`"?

Reply:
I think that it's worthwhile to both document that something _could be_ `X or None`, and to explain _why_ something might be None. The reason might not be obvious in some circumstances, particularly when a descriptive exception shouldn't be raised for each failure mode.

Reply:
That's very reasonable; I'll make sure it's documented clearly.
I will note, by way of explanation, that I'm not a big fan of properties which raise exceptions, which is why I chose the `X | None` paradigm for these.